index.js

```diff
@@ -174,16 +177,43 @@ Object.defineProperty(File.prototype, 'contents', {
   },
 });
 
+Object.defineProperty(File.prototype, 'cwd', {
+  get: function() {
+    return this._cwd;
+  },
+  set: function(cwd) {
+    if (typeof cwd !== 'string') {
+      throw new Error('cwd should be a string.');
+    }
+    this._cwd = endWithSep(normalize(cwd));
+  },
+});
+
+Object.defineProperty(File.prototype, 'base', {
+  get: function() {
+    return this._base || this._cwd;
+  },
+  set: function(base) {
+    if (base && typeof base !== 'string') {
+      throw new Error('base should be a string.');
+    }
+    this._base = base ? endWithSep(normalize(base)) : null;
```

Comment:
If `base` is falsey, return set `this._base` to null (or maybe delete) and return early; else reuse the `base` variable for the normalization and then compare before assigning to `this._base`